dist/s_evergreen

```diff
@@ -4,6 +4,35 @@
 setup_trap
 cd_top
 
+echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+
+echo "Check for Python dependencies without specified versions:"
+
+unspecified_dependency=false
+
+pip_installs=$(grep -E 'pip3? install' "test/evergreen.yml")
+
+while IFS= read -r line; do
+    for word in $line; do
+        if [[ "$word" =~ ^pip3?$ || "$word" == "install" || "$word" =~ ^python3?$ || "$word" =~ -.* ]]; then
+            continue
+        fi
+        # pip uses == to specify a specific version. Make sure these characters are present
+        if [[ ! "$word" =~ .*==.* ]]; then
+            unspecified_dependency=true
+            echo "Version not specified for Python dependency $word in line: $line"
+        fi
+    done
+done <<< "$pip_installs"
+
+if $unspecified_dependency; then
+    exit 1
```

Comment:
If we exit here we won't run `evg_cfg` below, so users won't see all errors from a single run of `s_all`.

Can we instead set an `error` flag and replace the `exit 0` at the end of the file with `exit $error`? 
Note that we want to capture errors from the `evg_cfg` checks below so that needs updating as well